src/NexusMods.App.UI/Pages/TextEdit/TextEditorPageViewModel.cs

```diff
@@ -0,0 +1,122 @@
+
+namespace NexusMods.App.UI.Pages.TextEdit;
+
+[UsedImplicitly]
+public class TextEditorPageViewModel : APageViewModel<ITextEditorPageViewModel>, ITextEditorPageViewModel
+{
+    [Reactive] public TextEditorPageContext? Context { get; set; }
+
+    [Reactive] public bool IsReadOnly { get; set; }
+
+    [Reactive] public bool IsModified { get; set; }
+
+    [Reactive] public TextDocument? Document { get; set; }
+
+    public ReactiveCommand<Unit, Unit> SaveCommand { get; }
+    private readonly ReactiveCommand<TextEditorPageContext, ValueTuple<TextEditorPageContext, string>> _loadFileCommand;
+
+    public TextEditorPageViewModel(
+        ILogger<TextEditorPageViewModel> logger,
+        IWindowManager windowManager,
+        IFileStore fileStore,
+        IConnection connection) : base(windowManager)
+    {
+        TabIcon = IconValues.FileDocumentOutline;
+        TabTitle = "Text Editor";
+
+        _loadFileCommand = ReactiveCommand.CreateFromTask<TextEditorPageContext, ValueTuple<TextEditorPageContext, string>>(async context =>
+        {
+            var fileId = context.FileId;
+
+            var fileHash = connection.Db.Get<StoredFile.Model>(fileId.Value).Hash;
+            logger.LogDebug("Loading file {Hash} into the Text Editor", fileHash);
+
+            await using var stream = await fileStore.GetFileStream(fileHash);
+            using var reader = new StreamReader(stream, Encoding.UTF8);
+            var contents = await reader.ReadToEndAsync();
+
+            return (context, contents);
+        });
+
+        var canSaveObservable = this.WhenAnyValue(
+            vm => vm.Document,
+            vm => vm.IsModified,
+            (document, isModified) => document is not null && isModified
+        );
+
+        SaveCommand = ReactiveCommand.CreateFromTask(async () =>
+        {
+            var fileId = Context!.FileId;
+            var fileName = Context!.FileName;
+
+            var text = Document?.Text ?? string.Empty;
+
+            // hash and store the new contents
+            var bytes = Encoding.UTF8.GetBytes(text);
+            var hash = Hash.From(XxHash64Algorithm.HashBytes(bytes));
+            var size = Size.From((ulong)bytes.Length);
+
+            var ms = new MemoryStream(bytes, writable: false);
+            var streamFactory = new MemoryStreamFactory(fileName, ms);
+
+            await fileStore.BackupFiles([new ArchivedFileEntry(streamFactory, hash, size)]);
+
+            // update the file
+            var db = connection.Db;
+            var storedFile = db.Get<StoredFile.Model>(fileId.Value);
+
+            using (var tx = connection.BeginTransaction())
+            {
+                tx.Add(storedFile.Id, StoredFile.Hash, hash);
+                tx.Add(storedFile.Id, StoredFile.Size, size);
+                storedFile.Remap<File.Model>().Mod.Revise(tx);
+                await tx.Commit();
+            }
+        }, canSaveObservable);
+
+        this.WhenActivated(disposables =>
+        {
```

Comment:
Would it be an idea to listen to the File Revisions from DB and reloading the page based on that?

This would keep two pages of the same document in synch if one makes changes and saves.

Reply:
Like this?

https://github.com/Nexus-Mods/NexusMods.App/assets/16577545/1244314e-c76f-4d32-b0b3-433730e74d26

Reply:
Cool, I guess the main issue would be if someone has both editors with changes, you are going to lose the progress of one of the two, but that is going to happen regardless. At least now it is clearer what is the latest version of the file.

Reply:
Works great with IsModified updates